src/query/atomWithQuery.ts

```diff
@@ -123,8 +130,15 @@ export function atomWithQuery<
         typeof createQuery === 'function' ? createQuery(get) : createQuery
 
       const defaultedOptions = queryClient.defaultQueryOptions(options)
-      const observer = new QueryObserver(queryClient, defaultedOptions)
-      const initialResult = observer.getCurrentResult()
+
+      if (observer) {
```

Comment:
This doesn't work if `queryClient` is changed, right?

Reply:
I cannot say yes or no! but in useQuery, it tracks the queryClient!